apps/reactotron-app/src/renderer/util/analyticsHelpers.tsx

```diff
@@ -0,0 +1,261 @@
+        })
+    }
+
+    if (!initialized) {
+      initialize()
+      setInitialized(true)
+    }
+  }, [initialized, optedOut])
+
+  // Send an analytics event
+  // This is the main function we use to send events throughout the app.
+  // See documentation here for how to use react-ga4:
+  // https://github.com/codler/react-ga4
+  const sendAnalyticsEvent = useCallback(
+    (event: UaEventOptions) => {
+      if (!optedOut) {
+        console.log("[analytics] Sending event", event)
+        ReactGA.event(event)
+      }
+    },
+    [optedOut]
+  )
+
+  // Send a page view event
+  const sendPageViewAnalyticsEvent = useCallback(
+    (page: string) => {
+      if (!optedOut) {
+        console.log("[analytics] Sending page view event", page)
+        ReactGA.send({ hitType: "pageview", page })
+      }
+    },
+    [optedOut]
+  )
+
+  // Send a keyboard shortcut event
+  const sendKeyboardShortcutAnalyticsEvent = useCallback(
+    (label: string) => {
+      sendAnalyticsEvent({
+        category: "navigation",
+        action: "keyboard_shortcut",
+        nonInteraction: false,
+        label,
+      })
+    },
+    [sendAnalyticsEvent]
+  )
+
+  // Send a custom command event
+  const sendCustomCommandAnalyticsEvent = useCallback(
+    (command: string, title: string) => {
+      sendAnalyticsEvent({
+        category: "custom_command",
+        action: command,
+        nonInteraction: false,
+        label: title,
+      })
+    },
+    [sendAnalyticsEvent]
+  )
+
+  // Send an external link event
+  const sendExternalLinkAnalyticsEvent = useCallback(
+    (label: string) => {
+      sendAnalyticsEvent({
+        category: "external_link",
+        action: "click",
+        nonInteraction: false,
+        label,
+      })
+    },
+    [sendAnalyticsEvent]
+  )
+
+  return {
+    initializeAnalytics,
+    sendAnalyticsEvent,
+    sendPageViewAnalyticsEvent,
+    sendKeyboardShortcutAnalyticsEvent,
+    sendCustomCommandAnalyticsEvent,
+    sendExternalLinkAnalyticsEvent,
+  }
+}
+
+// This hook is used to track page views within the app automatically using react-router-dom
+// This hook should only be used one time in the app, near the root of the component tree
+// but inside the HashRouter.
+export const usePageTracking = () => {
+  const location = useLocation()
+  const { sendPageViewAnalyticsEvent } = useAnalytics()
+
+  useEffect(() => {
+    sendPageViewAnalyticsEvent(location.pathname)
+  }, [location, sendPageViewAnalyticsEvent])
+}
+
+// This is a custom alert that we use to ask the user if they want to opt-in to analytics
+// We use this instead of the default alert because we want to style it to match our app.
+// We inherit the styles from react-confirm-alert and override them as needed.
+// Unfortunately, we can't use styled-components here because react-confirm-alert doesn't support it.
+// This also means we have to hard-code the colors here instead of using our theme, which is not ideal.
```

Comment:
just curious - how much is `react-confirm-alert` doing for us here? 
wondering if its worth the dependency if its difficult to style. could be nice to have full control of the UI, but might not be worth the effort.

Reply:
Yeah, when i put it in the app, i was thinking it would be useful as-is, but the more i played with it, the more customizations I wound up doing. I'll probably rip this out before i change this pr to no longer be a draft.